notebooks/Notebook2.md

```diff
@@ -2,33 +2,115 @@
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+
+I started this week by thinking a lot more about my DSL syntax as it pertains to characters and items within the game world. In this process, I made a lot of additions and changes. The first notable change is that I've removed the top-level rooms tag, and each room will sit at the top-level, with the items and characters within each room nested within. I made this change because I want to support multiple files combining into one game, and this makes it easier to write rooms in whatever file you want and have them merge together at compile time. To discern between characters and items, I've settled on the first option being 'item' or 'character' to tell these things apart. In addition, items can open doors or other items, can be grabbable (meaning it can go into your inventory or not), and can have a series of texts that print out upon looking, using, or picking them up. As for characters, you can specify any number of thing/text pairs - to be asked by the player in an 'ask character about thing' structure - by specifying the thing and the text separated by a colon. Finally, I've added start and end options for rooms to specify start and win states for the game. I also wrote in additional text for rooms conditional on some event occurring (like a key being used to open a door), but I'm not sure whether that will actually get implemented...
+
+Here's that expanded syntax below. There are still things to be sorted out that have been hinted in the comments at the top:
+
+```
+# Needs to support multiple items interacting on another item/door
+# Not quite satisfied with the blocked syntax... too imprecise
+# Should maybe revise 'opens' syntax to something more general like 'interacts with', but that's longer and can be a confusing descriptor in the case of locked doors
+
+
+<dark>(dark is start, treasure is north, cave is south, win is west) {
```

Comment:
This syntax reminds me of Baba is You. Do you know that game? Not fully related to your project, but some cool overlaps!